### PR TITLE
feat: approver ux UI

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
Well probably too many changes for 1 PR but oh well. 😅
- Implement Approver UX ui trying to keep the same Approval structure as on web
- Swap NavigationContainer with BottomSheetScrollView to unblock the scroll view on send sheet navigation
- Move Prism highlighting from extension to monorepo, unblock https://github.com/leather-io/extension/pull/5943
- Implement Prism highlighting in mobile

Initial route changed here to showcase the approver ux

https://github.com/user-attachments/assets/46e68aae-83e5-4171-bab0-6d5fe4e443b8

### CODE REVIEW

File: apps/mobile/src/features/approval-ux/approval.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approval-ux/approval.tsx:16]: The core styles for approver ux should be in UI package `src/components/approver`. 

Also approver should not be "aware" of any details like fees/inputs etc. `signPsbt` should be an instance of `<Approver />`

File: apps/mobile/src/features/approver/approver.tsx
Comment (kyranjamie) [inline apps/mobile/src/features/approver/approver.tsx:21]: Can we colocate this in `packages/ui/src/components/approver` though, or are you just working in mobile first and will migrate later for ease of dev?

Comment (kyranjamie) [reply apps/mobile/src/features/approver/approver.tsx:21]: for example, I think we can even reuse the same approver context for both implementations I think? As there's no platform specific behaviour inside (at least there shouldn't be)

File: apps/mobile/src/components/full-height-sheet/full-height-sheet.tsx
Comment (fbwoolf) [inline apps/mobile/src/components/full-height-sheet/full-height-sheet.tsx:17]: I made this change too locally, so came across the same issue. 👍

File: apps/mobile/src/features/approver/components/approver-account-card.tsx
Comment (fbwoolf) [inline apps/mobile/src/features/approver/components/approver-account-card.tsx:17]: Add these thru Crowdin?

File: apps/mobile/src/features/approver/components/approver-account-card.tsx
Comment (fbwoolf) [inline apps/mobile/src/features/approver/components/approver-account-card.tsx:21]: I made an `AccountBalance` and `AccountAddress` component in my PR that can go here at some point.

**Review by fbwoolf — APPROVED** [review]
This looks great @edgarkhanzadian. I may have created some conflicts for you bc I didn't realize you were moving stuff arnd with the send form routes. Let me know if you need help resolving.

File: packages/ui/src/components/approver/approver.native.tsx
Comment (kyranjamie) [inline packages/ui/src/components/approver/approver.native.tsx:19]: Could this be all wraped into its own hook so we share it between web/native?

**Review by kyranjamie — APPROVED** [review]

File: apps/mobile/src/features/approver/components/code-card.tsx
Comment (pete-watters) [reply apps/mobile/src/features/approver/components/code-card.tsx:14]: I think it would be good to put it in the UI library

File: apps/mobile/ios/leatherwalletmobile.xcodeproj/project.pbxproj
Comment (pete-watters) [inline apps/mobile/ios/leatherwalletmobile.xcodeproj/project.pbxproj:14]: I kept seeing this diff in my PR too, maybe it will go away once we merge this one?

File: apps/mobile/package.json
Comment (pete-watters) [inline apps/mobile/package.json:39]: I moved `@gorhom/bottom-sheet` dependancy into the UI library. 

Is there some reason not to use standard `Sheet`? Maybe we can export `BottomSheetScrollView` from the UI library and import here to avoid extra dependancies?

The same for `prism-react-renderer`, if we can move that component into the library then we could save the import here

File: apps/mobile/src/components/badges/badge.tsx
Comment (pete-watters) [inline apps/mobile/src/components/badges/badge.tsx:1]: Can this go into the UI library?

File: apps/mobile/src/components/badges/network-badge.tsx
Comment (pete-watters) [inline apps/mobile/src/components/badges/network-badge.tsx:1]: I'm not sure about having state connected components under `src/components` 

I would think they should be a feature?

File: apps/mobile/src/features/approver/components/approver-account-card.tsx
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/approver-account-card.tsx:14]: Maybe just load the first account from wallets here? Then it will be more realistic

File: apps/mobile/src/features/approver/components/inputs-outputs-card.tsx
Comment (pete-watters) [inline apps/mobile/src/features/approver/components/inputs-outputs-card.tsx:5]: How about just using the `<Flag` inline instead of having `UtxoRow`? 

Or else move it into it's own component

File: apps/mobile/src/features/send/send-sheets/sign-psbt.tsx
Comment (pete-watters) [inline apps/mobile/src/features/send/send-sheets/sign-psbt.tsx:15]: This could be it's own `ActionButton` component rather than a function?

**Review by pete-watters — APPROVED** [review]
Great work here @edgarkhanzadian 👍 

I added some organisation suggestions but really solid PR overall

File: apps/mobile/src/components/badges/network-badge.tsx
Comment (fbwoolf) [reply apps/mobile/src/components/badges/network-badge.tsx]: Yeah, we have some dep cruiser rules for this in the extension we need to add to mobile...

Screenshot 2024-11-15 at 1 36 49 PM

Comment (pete-watters) [reply apps/mobile/src/components/badges/network-badge.tsx]: My understanding is similar but that if a shared component is connected to state, it then becomes a `feature` rather than a `component`. That's just how I have seen things done in the past generally. 

I'm not sure if we can import one feature to another with those cruiser rules? Maybe they are enforcing a different setup we prefer. 

Usually only dumb, non state connected components would go to `src/components/` then state connected ones into `src/features`. If a component is used in multiple features:
- it could be made more dumb and receive props rather than hook into state
- if it needs to fetch state be promoted to a feature

**Pull request merged**
